### PR TITLE
 granite: 5.2.1 -> 5.2.2 

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
Closes #52882

Usual update but with stuff that I've already done to granite in https://github.com/NixOS/nixpkgs/pull/48637 that should be on master.
Most importantly this should fix certain elementary third-party applications from breaking.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build deepin.deepin-wm

File: pkgs/development/libraries/granite/default.nix [hunk 1]
```diff
@@ -1,38 +1,53 @@
...
     vala_0_40 # should be `elementary.vala` when elementary attribute set is merged
+    wrapGAppsHook
   ];
 
   buildInputs = [
     glib
-    gnome3.libgee
     gtk3
+    hicolor-icon-theme
+    gnome3.libgee
```
Comment (jtojnar) [inline pkgs/development/libraries/granite/default.nix]: Can be just `libgee` now.

File: pkgs/development/libraries/granite/default.nix [hunk 2]
```diff
@@ -1,38 +1,53 @@
...
+    (fetchpatch {
+      url = "https://github.com/elementary/granite/commit/2066b377226cf327cb2d5399b6b40a2d36d47b11.patch";
+      sha256 = "1bxjgq8wvl1sb79cwhmh9kwawnkkfn7c5q67cyz1fjxmamwyyi85";
+    })
+    (fetchpatch {
+      url = "https://github.com/elementary/granite/commit/f1b29f52e3aaf0f5d6bba44c42617da265f679c8.patch";
+      sha256 = "0cdp9ny6fj1lpcirab641p1qn1rbsvnsaa03hnr6zsdpim96jlvs";
+    })
+    # Resolve the circular dependency between granite and the datetime wingpanel indicator
+    ./02-datetime-clock-format-gsettings.patch
```
Comment (jtojnar) [inline pkgs/development/libraries/granite/default.nix]: Please add a link to the patch source.

File: pkgs/development/libraries/granite/default.nix (hunk 1 above)
Comment (worldofpeace) [reply pkgs/development/libraries/granite/default.nix]: That change isn't on master yet.

**Review by jtojnar — APPROVED** [review]

**Comment by worldofpeace** [comment]
Thanks @jtojnar

**Pull request merged**